server/chat-plugins/chatlog.ts

```diff
@@ -626,6 +663,183 @@ export abstract class Searcher {
+	// so it's easier to just do it the same way for both.
+	async roomStats(room: RoomID, month: string) {
+		if (!FS(`logs/chat/${room}`).existsSync()) {
+			return LogViewer.error(Utils.html`Room ${room} not found.`);
+		}
+		if (!FS(`logs/chat/${room}/${month}`).existsSync()) {
+			return LogViewer.error(Utils.html`Room ${room} does not have logs for the month ${month}.`);
+		}
+		const stats = await PM.query({
+			queryType: 'roomstats', search: month, roomid: room,
+		});
+		let buf = `<div class="pad"><h2>Room stats for ${room} [${month}]</h2><hr />`;
+		buf += `<strong>Total days with logs: ${stats.average.days}</strong><br />`;
+		const next = LogReader.nextMonth(month);
+		const prev = LogReader.prevMonth(month);
+		const prevExists = FS(`logs/chat/${room}/${prev}`).existsSync();
+		const nextExists = FS(`logs/chat/${room}/${next}`).existsSync();
+		if (prevExists) {
+			buf += `<br /><a roomid="view-roominfo-${room}--${prev}">Previous month</a>`;
+			buf += nextExists ? ` | ` : `<br />`;
+		}
+		if (nextExists) {
+			buf += `${prevExists ? `` : `<br />`}<a roomid="view-roominfo-${room}--${next}">Next month</a><br />`;
+		}
+		buf += this.visualizeStats(stats.average);
+		buf += `<hr />`;
+		buf += `<details class="readmore"><summary><strong>Stats by day</strong></summary>`;
+		for (const day of stats.days) {
+			buf += `<div class="infobox"><strong><a roomid="view-chatlog-${room}--${day.day}">${day.day}</a></strong><br />`;
+			buf += this.visualizeStats(day);
+			buf += `</div>`;
+		}
+		buf += '</details>';
+		return LogViewer.linkify(buf);
+	}
+	visualizeStats(stats: RoomStats) {
+		const titles: {[k: string]: string} = {
+			deadTime: 'Average time between lines',
+			deadPercent: 'Average % of the day spent more than 5m inactive',
+			linesPerUser: 'Average lines per user',
+			averagePresent: 'Average users present',
+			totalLines: 'Average lines per day',
+		};
+		let buf = `<div class="ladder pad"><table><tr><th>`;
+		buf += Object.keys(titles).map(k => titles[k]).join('</th><th>');
+		buf += `</th></tr><tr>`;
+		for (const k in titles) {
+			buf += `<td>`;
+			switch (k) {
+			case 'deadTime':
+				buf += Chat.toDurationString(stats.deadTime, {precision: 2});
+				break;
+
+			case 'linesPerUser': case 'totalLines': case 'averagePresent': case 'deadPercent':
+				const [main, dec] = (stats[k] || 0).toString().split('.');
+				buf += `${main}${dec ? `.${dec.slice(0, 2)}` : ``}`;
+				break;
+			}
+			buf += `</td>`;
+		}
+		buf += `</tr></table></div>`;
+		return buf;
+	}
+	async activityStats(room: RoomID, month: string) {
+		const days = (await FS(`logs/chat/${room}/${month}`).readdir()).map(f => f.slice(0, -4));
+		const stats: RoomStats[] = [];
+		for (const day of days) {
+			const curStats = await this.dayStats(room, day);
+			if (!curStats) continue;
+			stats.push(curStats);
+		}
+		// now, having collected the stats for each day, we need to merge them together
```

Comment:
We should probably add the stats to the total as we iterate over the days. This means we only have one loop.